### PR TITLE
bazel: update to 6.0.0rc1

### PR DESCRIPTION
**Pull request by keith**

### CODE REVIEW

File: ci/windows_ci_steps.sh [hunk 1]
```diff
@@ -88,7 +88,7 @@ fi
 
 # Complete envoy-static build
 if [[ $BUILD_ENVOY_STATIC -eq 1 ]]; then
-  bazel "${BAZEL_STARTUP_OPTIONS[@]}" build "${BAZEL_BUILD_OPTIONS[@]}" //source/exe:envoy-static
+  bazel "${BAZEL_STARTUP_OPTIONS[@]}" build --output_groups=+pdb_file "${BAZEL_BUILD_OPTIONS[@]}" //source/exe:envoy-static
```
Comment (keith) [inline ci/windows_ci_steps.sh:91]: New requirement to force bazel to download this when using remote exec

File: bazel/repository_locations.bzl [hunk 2]
```diff
@@ -1376,6 +1376,19 @@ REPOSITORY_LOCATIONS_SPEC = dict(
         release_date = "2022-03-30",
         cpe = "N/A",
     ),
+    rules_license = dict(
```
Comment (keith) [inline bazel/repository_locations.bzl:1379]: New internal bazel dep which shows up in the dependency validation queries

Comment (phlax) [reply bazel/repository_locations.bzl:1379]: wow - cool addition - we had started to add something similar - im wondering if/how we can hook  this in

Comment (keith) [reply bazel/repository_locations.bzl:1379]: We definitely can

File: ci/windows_ci_steps.sh (hunk 1 above)
Comment (phlax) [reply ci/windows_ci_steps.sh:91]: i dont entirely understand why this is now required here but wondering if we need to add it anywhere else and/or update relevant docs

Comment (phlax) [reply ci/windows_ci_steps.sh]: cool, thanks for explanation

**Review by phlax — APPROVED** [review]
lgtm, thanks @keith

File: bazel/repository_locations.bzl [hunk 3]
```diff
@@ -1376,6 +1376,19 @@ REPOSITORY_LOCATIONS_SPEC = dict(
...
+        project_name = "rules_license",
+        project_desc = "Bazel rules for checking open source licenses",
+        project_url = "https://github.com/bazelbuild/rules_license",
+        version = "0.0.3",
+        sha256 = "00ccc0df21312c127ac4b12880ab0f9a26c1cff99442dc6c5a331750360de3c3",
+        urls = ["https://github.com/bazelbuild/rules_license/releases/download/0.0.3/rules_license-0.0.3.tar.gz"],
```
Comment (moderation) [inline bazel/repository_locations.bzl:1385]: I didn't see this one come through - we shouldn't hard code the versions here. Should be
`        urls = ["https://github.com/bazelbuild/rules_license/releases/download/{version}/rules_license-{version}.tar.gz"],`

Comment (keith) [reply bazel/repository_locations.bzl:1385]: oops thanks! https://github.com/envoyproxy/envoy/pull/23780

**Pull request merged**